crates/csharp/src/lib.rs

```diff
@@ -635,20 +657,80 @@ impl InterfaceGenerator<'_> {
                         return BitConverter.ToUInt16(span);
                     }}
 
-                    public static int GetS32(IntPtr ptr)
+                    internal static int GetS32(IntPtr ptr)
                     {{
                         var span = new Span<byte>((void*)ptr, 4);
 
                         return BitConverter.ToInt32(span);
                     }}
 
-                    public static string GetUTF8String(IntPtr ptr)
+                    internal static long GetS64(IntPtr ptr)
+                    {{
+                        var span = new Span<byte>((void*)ptr, 8);
+
+                        return BitConverter.ToInt64(span);
+                    }}
+
+                    internal static float GetF32(IntPtr ptr)
+                    {{
+                        var span = new Span<byte>((void*)ptr, 4);
+
+                        return BitConverter.ToSingle(span);
+                    }}
+
+                    internal static void SetS16(IntPtr ptr, short value)
+                    {{
+                        var span = new Span<byte>((void*)ptr, 2);
+
+                        BitConverter.TryWriteBytes(span, value);
+                    }}
+
+                    internal static void SetS32(IntPtr ptr, int value)
+                    {{
+                        var span = new Span<byte>((void*)ptr, 4);
+
+                        BitConverter.TryWriteBytes(span, value);
+                    }}
+
+                    internal static void SetU32(IntPtr ptr, uint value)
+                    {{
+                        var span = new Span<byte>((void*)ptr, 4);
+
+                        BitConverter.TryWriteBytes(span, value);
+                    }}
+
+                    internal static void SetArray<T>(IntPtr ptr, T[] value, int length)
+                    {{
+                        var span = new Span<byte>((void*)ptr, sizeof(T)*length);
+
+                        fixed (T* dstPtr = &value[0])
+                        {{
+                            Marshal.Copy(span.ToArray(), 0, (IntPtr)dstPtr, length);
+                        }}
+                    }}
+
+                    unsafe internal static T[] GetArray<T>(IntPtr ptr, int length)
+                    {{
+                        var span = new Span<byte>((void*)ptr, sizeof(T)*length);
+        
+                        var array = new T[length];
+        
+                        fixed (T* dstPtr = &array[0])
+                        {{
+                            //TODO: Optimize this. The ToArray call is not needed and does an additional heap allocation.
+                            Marshal.Copy(span.ToArray(), 0, (IntPtr)dstPtr, length);
```

Comment:
maybe you can use `Copy(nint source, byte[] destination, int startIndex, int length);` 
without the need for `var span = new Span<byte> ... ` and `fixed` at all

Reply:
I ended up with `span.CopyTo(new Span<T>(array));`  I believe this addresses the additional heap allocation.  I kept Span<T> since it was useful for handling all the different lengths of types